src/main/scala/org/kframework/kore/Interface.scala

```diff
@@ -0,0 +1,297 @@
+package org.kframework.kore
+
+trait Definition {
+  def modules: Seq[Module]
+  def att: Attributes
+}
+
+object Definition {
+  def unapply(arg: Definition): Option[(Seq[Module], Attributes)] = Some(arg.modules, arg.att)
+}
+
+trait Module {
+  def name: ModuleName
+  def sentences: Seq[Sentence]
+  def att: Attributes
+}
+
+object Module {
+  def unapply(arg: Module): Option[(ModuleName, Seq[Sentence], Attributes)] = Some(arg.name, arg.sentences, arg.att)
+}
+
+trait Sentence {
+  def att: Attributes
```

Comment:
This deviates from the translation rules at https://github.com/kframework/kore/wiki/KORE-Language-Syntax...